support/allowed_lints.txt

```diff
@@ -1,3 +1,5 @@
 clippy::module_inception
 clippy::new_ret_no_self
 clippy::new_without_default
+clippy::suspicious_else_formatting
+unused_must_use
```

Comment:
I think you just need to put these allows in https://github.com/davidMcneil/configopt/blob/master/configopt-derive/src/configopt_type/generate.rs and then fork that. That will isolate the exceptions to only ConfigOpt generated code.

Reply:
We will need to update our habitat-sh/structopt code as well to fix the clippy::suspicious_else_formatting linter error